boxes/boxes/vanilla/app/embedded-wallet.ts

```
@@ -4,71 +4,98 @@ import {
 import { randomBytes } from '@aztec/foundation/crypto';
-import { getEcdsaRAccount } from '@aztec/accounts/ecdsa/lazy';
-import { getSchnorrAccount } from '@aztec/accounts/schnorr/lazy';
+import { EcdsaRAccountContract } from '@aztec/accounts/ecdsa/lazy';
+import { SchnorrAccountContract } from '@aztec/accounts/schnorr/lazy';
+
 import { getPXEServiceConfig } from '@aztec/pxe/config';
 import { createPXEService } from '@aztec/pxe/client/lazy';
+import { getInitialTestAccountsData } from '@aztec/accounts/testing/lazy';
 import {
-  type ContractArtifact,
-  getDefaultInitializer,
-} from '@aztec/stdlib/abi';
-import { getInitialTestAccounts } from '@aztec/accounts/testing';
+  getStubAccountContractArtifact,
+  createStubAccount,
+} from '@aztec/accounts/stub/lazy';
+import { DefaultMultiCallEntrypoint } from '@aztec/entrypoints/multicall';
+import { ExecutionPayload } from '@aztec/entrypoints/payload';
+import { TxProvingResult, TxSimulationResult } from '@aztec/stdlib/tx';
 
 const PROVER_ENABLED = true;
 
 const logger = createLogger('wallet');
 const LocalStorageKey = 'aztec-account';
 
-// This is a minimal implementation of an Aztec wallet, that saves the private keys in local storage.
-// This does not implement `@aztec.js/Wallet` interface though
-// This is not meant for production use
-export class EmbeddedWallet {
-  private pxe!: PXE;
-  connectedAccount: AccountWallet | null = null;
+// This is a minimal implementation of an Aztec wallet
+// WARNING: This example code stores the wallet in plain text in LocalStorage. Do not use in production without understanding the security implications
+export class EmbeddedWallet extends BaseWallet {
+  connectedAccount: AztecAddress | null = null;
+  protected accounts: Map<string, Account> = new Map();
+
+  protected async getAccountFromAddress(
+    address: AztecAddress
+  ): Promise<Account> {
+    let account: Account | undefined;
+    if (address.equals(AztecAddress.ZERO)) {
```

I would split the function into `createAccount` and `getAccount`.

Having these ifs hidden in the function implementation that make it do something kind of random just result in bad code clarity. Fetching by invalid address should throw an error.

---

This is not random.

`from` is a mandatory param, but sometimes there's really no `from`...for example when deploying your own account contract, which does an `universalDeployment`. Essentially this gives the wallet the ability to send "anonymous" txs (in the sense there's no account contract involved)